scipy/optimize/optimize.py

```diff
@@ -1465,7 +1465,8 @@ def _minimize_newtoncg(fun, x0, args=(), jac=None, hess=None, hessp=None,
             A = fhess(*(xk,) + args)
             hcalls = hcalls + 1
 
-        while numpy.add.reduce(numpy.abs(ri)) > termcond:
+        k2 = 0
+        while (numpy.add.reduce(numpy.abs(ri)) > termcond) and (k2 < maxiter):
```

Comment:
I found MATLAB's defaults for `maxit` in CG: http://www.mathworks.com/help/matlab/ref/pcg.html, so it's `min(20, n)`. Can't really comment about this 20, should be some reason for that.